katello/katello-change-hostname

```diff
@@ -50,108 +124,105 @@ def yesno
+               "You will need to re-register any #{@options[:program]} clients that are registered to this server. " \
+               "When re-registering clients, you will have to reinstall the bootstrap RPM. \n"
 
-  if has_disable_system_check_option?
-    opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
-      options[:system_check] = true
-    end
+  if @foreman_proxy_content
+    STDOUT.print "IMPORTANT: You will have to update the Name and URL of the Smart Proxy in #{@options[:program].capitalize} to the new hostname.\n"
+  else
+    STDOUT.print "
+    For #{@plural_proxy}, you will need to regenerate the certs on the #{@options[:scenario].capitalize} server and reinstall the #{@proxy}
+    If you want to use custom certificates, re-run the #{@options[:program]}-installer with custom certificate options
+    Short hostnames have not been updated, please update those manually.\n"
   end
+  STDOUT.print ""
+  exit(true)
+end
 
-  opt.on("-h","--help","help") do
-    puts opt_parser
-    exit
+def warning_message
+  STDOUT.print("***WARNING*** This script will modify your system. " \
+               "You will need to re-register any #{@options[:program]} clients registered to this system after script completion.")
+  unless @foreman_proxy_content
+    STDOUT.print(" #{ @plural_proxy } will have to be re-registered and reinstalled. If you are using custom certificates, you " \
+                 "will have to run the #{@options[:program]}-installer again with custom certificate options after this script completes.")
   end
+  STDOUT.print(" Have you taken the necessary precautions (backups, snapshots, etc...) and want to proceed with " \
+               "changing your hostname? \n [y/n]:")
 end
-opt_parser.parse!
 
-# kept as variables for easy changing in backports
-proxy = "Foreman Proxy"
-plural_proxy = "Foreman Proxies"
-
-STDOUT.print("This script will modify your system. You will need to re-register any #{ plural_proxy } and " \
-             "#{options[:scenario].capitalize} clients after script completion. #{ plural_proxy } " \
-             "will have to be reinstalled. If you are using custom certificates, you will have to run the " \
-             "#{options[:program]}-installer again with custom certificate options after this script completes." \
-             " Have you taken the necessary precautions (backups, snapshots, etc...) and want to proceed with " \
-             "changing your hostname? \n [y/n]:")
+STDOUT.print(warning_message)
 response = yesno
 unless response
-  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
-  exit(false)
+  fail_with_message("Hostname change aborted, no changes have been made to your system")
 end
 
-unless options[:username] && options[:password]
-  STDOUT.puts "Username and/or Password options are missing!"
-  puts opt_parser
-  exit(false)
+unless @options[:username] && @options[:password]
```

Comment:
There are a lot of "conditional" checks sprinkled throughout this lower section that can cause the script to bail out. They might be nice to consolidate into on area of the script, within a function or within a small module/class for readability and clarity.